### DISCUSSION TITLE
HELP!  Trying to Link Meeting Guide Data between Orange County Central Offices

### DISCUSSION THREAD

**Discussion by fbarnese** [code4recovery/12-step-meeting-list#1058] [Q&A]
We are using version 3.14.12; TMSL IU

The Orange County central offices share a meeting database.  Rather than pointing the www.aanoc.org to the meeting page at www.oc-aa.org, I am trying to link the data with the "Link Meeting Guide" option in the program.

Orange County Central Office - https://www.oc-aa.org/meetings - Parent Region - Change Detection Enabled.

Error:
Invalid response,
WP_Error Object
(
    [errors] => Array
        (
            [http_request_failed] => Array
                (
                    [0] => cURL error 28: Connection timed out after 10001 milliseconds
                )

        )

    [error_data] => Array
        (
        )

    [additional_data:protected] => Array
        (
        )

)
.

**Comment by joshreisner** [comment]
hi @fbarnese i'm going to move this to our TSML discussions, i want to put it somewhere that will help future users and i think they are most likely to look there. one sec

**Comment by joshreisner** [comment]
hi @fbarnese to answer your question, hopefully, it sounds like you might not have entered the right thing as your data source - i think that's on us because we don't have any labels on these fields! but it's this one:

![import](https://user-images.githubusercontent.com/1551689/213261907-410aaccc-c95f-4c0c-ab19-2e942b10d82e.png)

this needs to be a JSON feed address. we're considering changing that if/when we do away with feed security, but that's another matter.

i checked and oc-aa.org has feed security set to open, which is great. to import out of oc-aa.org and into aanoc.org,  on the Meetings > import / Export page of aanoc.org you want to put the address `https://oc-aa.org/wp-admin/admin-ajax.php?action=meetings`

(this is the "Public Data Source" link which they can find at Meetings > Settings)

**Reply by joshreisner** [reply]
@fbarnese sounds like your web hosting company may have added a firewall to your site, you might try asking them to disable it, or perhaps "whitelist" the `oc-aa.org` domain for you for outbound requests

firewalls are pretty common on shared hosting plans as a spam-prevention technique

**Comment by fbarnese** [comment]
Thanks for the help Josh.  However, it is still not allowing me to add the feed ... even with the json address.  Error below:


Invalid response,

WP_Error Object
(
    [errors] => Array
        (
            [http_request_failed] => Array
                (
                    [0] => cURL error 28: Connection timed out after 10001 milliseconds
                )

        )

    [error_data] => Array
        (
        )

    [additional_data:protected] => Array
        (
        )

)


.

________________________________

**Comment by fbarnese** [comment]
Hi Josh.

The North Orange County website will only upload about 25 records and then stalls.  I've disabled the firewall and it still doesn't move closer to the 2089 meetings that we need to share.
Any other thoughts on this?

Faith
________________________________

**Reply by joshreisner** [reply]
2089 - that's quite a lot of meetings! 

so sounds like you were able to resolve the firewall issue from before?

so you're trying to import out of `aanoc.org` into `oc-aa.org` or is it vice-versa? we can try it on our end to see if it's a data issue, but my guess is it's probably some kind of javascript error when you're importing

could you try [opening the javascript console](https://www.coursera.support/s/article/learner-000001653-How-to-open-the-Javascript-console?language=en_US) and uploading a screenshot of what you see during the import?

you will have to come to github to do that, email attachment images don't show up here. thanks

**Comment by fbarnese** [comment]
Export from oc-aa.org into aanoc.org.
I will upload the image to the javascript console.

Faith
________________________________

**Comment by fbarnese** [comment]
<img width="2309" alt="Screenshot 2023-01-26 at 12 56 39 PM" src="https://user-images.githubusercontent.com/123008919/214949019-262037c9-b60f-482d-8e83-ec71f798d3c4.png">

**Reply by joshreisner** [reply]
looks like "Logs" is selected, could you change that to "Errors" or "All" and refresh and rerun the import? thanks! 

i just tried it on our site and it worked so i don't think it's a problem on `oc-aa.org` or with the software under normal conditions

![code4recovery](https://user-images.githubusercontent.com/1551689/214949411-7acb0b96-c8b5-4255-b6fd-f2d99f0cee2e.png)

**Comment by fbarnese** [comment]
<img width="2145" alt="Screenshot 2023-01-26 at 1 05 05 PM" src="https://user-images.githubusercontent.com/123008919/214950388-1be5537d-eeaa-432d-9deb-ae35430afac2.png">

**Reply by joshreisner** [reply]
hmmmmm, thanks! not sure what's going on here.

i notice you already have a bunch of meetings on your site, were they uploaded by a CSV or imported by JSON and it stopped working?

**Reply by fbarnese** [reply]
I've tried both ways.  The CSV stops at 6% and I don't see that the import goes any further.

**Reply by fbarnese** [reply]
Also, I current have the aanoc.org website pointing to the oc-aa.org meeting guide so we can service the membership.  However, we end up losing contributions to our sister office so I need to get these sites their own versions of the same meeting file.

**Reply by joshreisner** [reply]
the importer works in batches of 25, so sounds like it's dying after the first batch finishes. not sure why!

i wonder if you'd be open to trying another idea. do you not need to make any edits to the meetings locally? (like are they all made at oc-aa.org)? if so, then you could probably just use TSML UI without importing.

try creating a new page on your site and embedding this HTML code:

```
<div
  id="tsml-ui"
  data-src="https://oc-aa.org/wp-admin/admin-ajax.php?action=meetings"
  data-mapbox="pk.eyJ1IjoiZGF2ZWc4MSIsImEiOiJja3IwcnlhZGoxdjlxMzFxaHNydDBza3ByIn0.t-gsVDq-_V5cl_0rW13y6Q"
  data-timezone="America/Los_Angeles"
></div>
<script src="https://tsml-ui.code4recovery.org/app.js" async></script>
```

here's an explainer on how to add html code: https://www.inmotionhosting.com/support/edu/wordpress/adding-html-wordpress/

not sure if there's a better way with elementor if you're using that. definitely see if there are any template options to create the max space on the page, aka full page mode

the advantage of this approach is you won't have to do anything on `aanoc.org` and you should get the latest listings from `oc-aa.org`

**Reply by lovethecodelabs** [reply]
you may benefit from increasing the PHP Memory available to Wordpress, to run the importer correctly.

**Comment by fbarnese** [comment]
I've created a new "TEST Meeting Directory: page and added the html code.  It is not pulling any data to populate the page ... this tells me that somethings is still blocking it on the aanoc.org site.  (ugh.)
________________________________

**Reply by fbarnese** [reply]
https://www.aanoc.org/meeting-directory/

**Reply by joshreisner** [reply]
i'm sorry i pasted the wrong code, then updated it, but edits don't show up in emails :(

could you update it to:

```
<div
  id="tsml-ui"
  data-src="https://oc-aa.org/wp-admin/admin-ajax.php?action=meetings"
  data-mapbox="pk.eyJ1IjoiZGF2ZWc4MSIsImEiOiJja3IwcnlhZGoxdjlxMzFxaHNydDBza3ByIn0.t-gsVDq-_V5cl_0rW13y6Q"
  data-timezone="America/Los_Angeles"
></div>
<script src="https://tsml-ui.code4recovery.org/app.js" async></script>
```

the key part is at the bottom, the `https://tsml-ui.code4recovery.org/app.js`

**Reply by joshreisner** [reply]
ha looks like it's working!! you could theoretically remove the 12 Step Meeting List plugin now…

**Reply by joshreisner** [reply]
if you wanted to remove that redundant page title you could add this CSS:

```
body.page-id-64554 .page-header { display: none; }
```

**Comment by fbarnese** [comment]
I will remove the plugin ... don't want my team to be confused.
Thank you so very much Josh.  You are a gem!

Faith
________________________________

**Comment by fbarnese** [comment]
Hi Josh.

I have another question that would be easier to explain if we were to chat.  Is it possible for you to call me?  (480) 532-0007

Faith Barnese
________________________________
